src/test/java/org/apache/commons/io/performance/IOUtilsContentEqualsPerformanceTest.java

```diff
@@ -0,0 +1,178 @@
+                 Reader inputReader1 = new InputStreamReader(inputStream1);
+                 Reader inputReader2 = new InputStreamReader(inputStream2);
+            ) {
+                res[1] = contentEqualsOld(inputReader1, inputReader2);
+            }
+        }
+        return res;
+    }
+
+    @Benchmark
+    public boolean[] testContentEqualsNew2() throws IOException {
+        boolean[] res = new boolean[9];
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                try (Reader inputReader1 = new StringReader(STRINGS[i]);
+                     Reader inputReader2 = new StringReader(STRINGS[j]);
+                ) {
+                    res[i * 3 + j] = IOUtils.contentEquals(inputReader1, inputReader2);
+                }
+            }
+        }
+        return res;
+    }
+
+    @Benchmark
+    public boolean[] testContentEqualsOld2() throws IOException {
+        boolean[] res = new boolean[9];
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                try (Reader inputReader1 = new StringReader(STRINGS[i]);
+                     Reader inputReader2 = new StringReader(STRINGS[j]);
+                ) {
+                    res[i * 3 + j] = contentEqualsOld(inputReader1, inputReader2);
+                }
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Old version of IOUtils.contentEquals(Reader, Reader)
+     *
+     * Compares the contents of two Readers to determine if they are equal or
+     * not.
+     * <p>
+     * This method buffers the input internally using
+     * <code>BufferedReader</code> if they are not already buffered.
+     * </p>
+     *
+     * @param input1 the first reader
+     * @param input2 the second reader
+     * @return true if the content of the readers are equal or they both don't
+     * exist, false otherwise
+     * @throws NullPointerException if either input is null
+     * @throws IOException          if an I/O error occurs
+     * @since 1.1
+     */
+    @SuppressWarnings("resource")
+    public static boolean contentEqualsOld(final Reader input1, final Reader input2)
+            throws IOException {
+        if (input1 == input2) {
+            return true;
+        }
+        if (input1 == null ^ input2 == null) {
+            return false;
+        }
+        final BufferedReader bufferedInput1 = toBufferedReader(input1);
+        final BufferedReader bufferedInput2 = toBufferedReader(input2);
+
+        int ch = bufferedInput1.read();
+        while (EOF != ch) {
+            final int ch2 = bufferedInput2.read();
+            if (ch != ch2) {
+                return false;
+            }
+            ch = bufferedInput1.read();
+        }
+
+        return bufferedInput2.read() == EOF;
+    }
+
+    public static String getString0() {
+        StringBuilder stringBuilder = new StringBuilder("ab");
+        for (int i = 0; i < 24; i++) {
```

Comment:
Did you test with larger strings?
24 looks like a small string

Reply:
@eolivelli
stringBuilder.append(stringBuilder);
that is 2^24, not 2*24...
And I think that is long enough.
If I use lang3.StringUtils.repeat maybe it will be more clear, but I'm not quite sure whethere it be good to use it in commons-io.

Reply:
Got it.
Sorry for the confusion